Exscript/protocols/protocol.py

```diff
@@ -282,6 +286,8 @@ def __init__(self,
         self.banner_timeout = banner_timeout
         self.encoding = encoding
         self.send_data = None
+        self.auth_methods = auth_methods
```

Comment:
I would do a sanity check here, i.E.:

```
for method in auth_methods:
    if method not in auth_types:
        raise ValueError('unsupported auth_method:', method)
```

Reply:
I thought about this check in the beginning. However the *auth_type* dict is in the *ssh2.py* module while this should be done in module *protocol.py*. If we import *ssh2* in *protocol* we will end up with a circular import.

Another solution would be to have the auth_method attribute set directly inside *ssh2* and not in *protocol*

Reply:
I would leave "self.auth_methods = auth_methods" in the protocol adapter, but only check the sanity in the SSH2 adapter. That would allow for code that can be interchanged between Telnet and SSH.